### PR TITLE
NuGet: Scopes and direct dependencies

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Add support for separate scopes and an option to resolve only direct dependencies to NuGet based package managers. See the commit messages for details.

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt [hunk 1]
```diff
@@ -304,25 +304,47 @@ fun PackageManager.resolveNuGetDependencies(
...
     val references = reader.getDependencies(definitionFile)
-    val ids = references.map { Identifier(type = "NuGet", namespace = "", name = it.name, version = it.version) }
-    support.buildDependencyTree(ids, dependencies, packages, issues)
+    val referencesByFramework = references.groupBy { it.targetFramework }
+
+    val scopes = referencesByFramework.flatMapTo(sortedSetOf()) { (targetFramework, frameworkDependencies) ->
+        frameworkDependencies.groupBy { it.developmentDependency }.map { (isDevDependency, dependencies) ->
+            val allDependencies = buildSet {
+                addAll(dependencies)
+                // Add dependencies without a specified target framework to all scopes.
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt:325]: More precise: "Add all *development* dependencies..."

File: analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt [hunk 2]
```diff
@@ -304,25 +304,47 @@ fun PackageManager.resolveNuGetDependencies(
...
+                addAll(referencesByFramework[""].orEmpty().filter { it.developmentDependency == isDevDependency })
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt:326]: For readability, `referencesByFramework[""].orEmpty()` could be extracted to a variable before the loops, e.g. to line 312 like `val dependenciesForAllFrameworks = referencesByFramework[""].orEmpty()`.

File: analyzer/src/main/kotlin/managers/DotNet.kt [hunk 3]
```diff
@@ -62,7 +62,14 @@ class DotNet(
     private val reader = DotNetPackageFileReader()
 
     override fun resolveDependencies(definitionFile: File, labels: Map<String, String>): List<ProjectAnalyzerResult> =
-        listOf(resolveNuGetDependencies(definitionFile, reader, NuGetSupport.create(definitionFile)))
+        listOf(
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/DotNet.kt:72]: Should we add a hint to the commit message that this is useful to match the behavior of tools like https://help.sonatype.com/iqserver/analysis/nuget-application-analysis#NuGetApplicationAnalysis-Evaluation:Projectfiles for easier comparison of results?

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt (hunk 1 above)
Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt:326]: Only if `isDevDependency` is true, otherwise it's normal dependencies.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**